volatility3/framework/plugins/linux/netfilter.py

```diff
@@ -0,0 +1,716 @@
+            layer_name=self.layer_name,
+            count=nf_hook_entries.num_hook_entries,
+        )
+
+        return orig_ops
+
+    def subscribed_protocols(self) -> Tuple[str]:
+        """Allows to select which PROTO_HOOKS protocols will be processed by the
+        Netfiler subclass.
+        """
+
+        # Most implementation handlers respond to these protocols, except for
+        # the ingress hook, which specifically handles the 'NETDEV' protocol.
+        # However, there is no corresponding Netfilter hook implementation for
+        # the INET protocol in the kernel. AFAIU, this is used as
+        # 'NFPROTO_INET = NFPROTO_IPV4 || NFPROTO_IPV6'
+        # in other parts of the kernel source code.
+        return ("IPV4", "ARP", "BRIDGE", "IPV6", "DECNET")
+
+    def get_module_name_for_address(self, addr) -> str:
+        """Helper to obtain the module and symbol name in the format needed for the
+        output of this plugin.
+        """
+        module_name, symbol_name = linux.LinuxUtilities.lookup_module_address(
+            self.vmlinux, self.handlers, addr
+        )
+
+        if module_name == "UNKNOWN":
+            module_name = None
+
+        if symbol_name != "N/A":
+            module_name = f"[{symbol_name}]"
+
+        return module_name
+
+    def get_net_namespaces(self):
+        """Common function to retrieve the different namespaces.
+        From 4.3 on, all the implementations use network namespaces.
+        """
+        nethead = self.vmlinux.object_from_symbol("net_namespace_list")
+        symbol_net_name = self.get_symbol_fullname("net")
+        for net in nethead.to_list(symbol_net_name, "list"):
+            net_ns_id = net.ns.inum
+            yield net_ns_id, net
+
+    def get_hooks_container(self, net, proto_name, hook_name):
+        """Returns the data structure used in a specific kernel implementation to store
+        the hooks for a respective namespace and protocol.
+
+        Except for kernels < 4.3, all the implementations use network namespaces.
+        Also the data structure which contains the hooks, even though it changes its
+        implementation and/or data type, it is always in this location.
+        """
+        yield net.nf.hooks
+
+    def get_hook_ops(self, hook_container, proto_idx, hook_idx):
+        """Given the hook_container obtained from get_hooks_container(), it
+        returns an iterable of 'nf_hook_ops' elements for a corresponding protocol
+        and hook type.
+
+        This is the most variable/unstable part of all Netfilter hook designs, it
+        changes almost in every single implementation.
+        """
+        raise NotImplementedError("You must implement this method")
+
+    def get_symbol_fullname(self, symbol_basename: str) -> str:
+        """Given a short symbol or type name, it returns its full name"""
+        return self.vmlinux.symbol_table_name + constants.BANG + symbol_basename
+
+    @staticmethod
+    def get_member_type(
```

Comment:
This feels a little black magicy...  You're definitely playing around with volatility internals here.  My concern is if there's a change and we add another object with a subtype or something.  I can see it gets used a lot and looks useful, but I worry about how robust it is.  If you're happy with it, I'll go for it, it's just got my spidey sense tingling a little...  5;)

Reply:
Well, so far I think it's handy as it's like a shortcut to avoid repeating the same formula (x+y+z) everywhere in the code. 
The good thing is that if your concern becomes true in the future, we hopefully need to modify this in just one place

Reply:
Well, this probably does need extracting, maybe adding as a method alongside [has_member](https://github.com/volatilityfoundation/volatility3/blob/develop/volatility3/framework/interfaces/objects.py#L279)?  That would let each type override it, but I'm wondering how useful it would be for most types, since it only really applies to structs?